### PR TITLE
Remove most uses of `usize` and `isize`

### PR DESCRIPTION
**Pull request by tbu-**
Done in two steps, first remove `is` and `us` suffixes which weren't needed, and then convert replace most uses of `usize` with `u32` and almost all uses of `isize` with `i32`.

### CODE REVIEW

File: src/blist.rs [hunk 1]
```diff
@@ -732,15 +732,15 @@ mod bench{
 
     #[bench]
     fn bench_collect_into(b: &mut test::Bencher) {
-        let v = &[0is; 64];
+        let v = &[0i32; 64];
```
Comment (Gankro) [inline src/blist.rs]: Is this annotation necessary?

File: src/blist.rs [hunk 2]
```diff
@@ -732,15 +732,15 @@ mod bench{
...
         b.iter(|| {
-            let _: BList<isize> = v.iter().map(|x| *x).collect();
+            let _: BList<i32> = v.iter().map(|x| *x).collect();
         })
     }
 
     #[bench]
     fn bench_push_front(b: &mut test::Bencher) {
-        let mut m: BList<isize> = BList::new();
+        let mut m: BList<i32> = BList::new();
```
Comment (Gankro) [inline src/blist.rs]: These all seem inferrable? Is the bencher infrastructure messing with this?

File: src/proto/dlist.rs [hunk 3]
```diff
@@ -1190,15 +1190,15 @@ mod bench{
...
         b.iter(|| {
-            let _: DList<isize> = v.iter().map(|x| *x).collect();
+            let _: DList<i32> = v.iter().map(|x| *x).collect();
         })
     }
 
     #[bench]
     fn bench_push_front(b: &mut test::Bencher) {
-        let mut m: DList<isize> = DList::new();
+        let mut m: DList<i32> = DList::new();
```
Comment (Gankro) [inline src/proto/dlist.rs]: Inferables?

File: src/proto/shootout.rs [hunk 4]
```diff
@@ -7,15 +7,15 @@ mod std_dlist_bench{
 
     #[bench]
     fn bench_collect_into(b: &mut test::Bencher) {
-        let v = &[0is; 64];
+        let v = &[0i32; 64];
```
Comment (Gankro) [inline src/proto/shootout.rs]: inferrable?

File: src/tree/set.rs [hunk 5]
```diff
@@ -1138,22 +1138,16 @@ mod test {
...
-        // FIXME: #5801: this needs a type hint to compile...
-        let result: Option<(&usize, & &'static str)> = z.next();
-        assert_eq!(result.unwrap(), (&5us, &("bar")));
-
-        let result: Option<(&usize, & &'static str)> = z.next();
-        assert_eq!(result.unwrap(), (&11us, &("foo")));
-
-        let result: Option<(&usize, & &'static str)> = z.next();
-        assert!(result.is_none());
+        assert_eq!(z.next().unwrap(), (&5, &("bar")));
```
Comment (Gankro) [inline src/tree/set.rs]: Nice!

**Comment by Gankro** [comment]
This looks awesome! One minor concern is that a lot of our benchmarks were built to mirror `std` benches for comparison. So this is basically going to semantically break a lot of our benches until `std` is changed to match. Not a big deal, though.

File: src/blist.rs (hunk 2 above)
Comment (tbu-) [inline src/blist.rs]: collect/src/blist.rs:744:11: 746:11 error: unable to infer enough type information about `closure[collect/src/blist.rs:744:16: 746:10]`; type annotations required

**Comment by tbu-** [comment]
@Gankro There's some weird interaction with unboxed closures and integer inference, it doesn't seem to completely work. I didn't check the benches explicitely, but they looked like other cases that broke (e.g. the one from your first comment).

**Comment by Gankro** [comment]
Okay, cool! Should be good to land now, can refine later if we want.

**Comment by Gankro** [comment]
*goes to run new benches and brag about how collect is now twice as fast as std in micro-benches*

**Comment by tbu-** [comment]
@Gankro Is the development of benches saved somewhere?

**Comment by Gankro** [comment]
@tbu- no, we don't publicly register the bench results anywhere, although that might be interesting to do.

**Comment by Gankro** [comment]
Unfortunately pervasive codegen issues make bench results a bit questionable (is this code better because it's better, or because rustc happens to shit the bed less with it?)

**Pull request merged**